test/integration/QueryRunnerTest.kt

```diff
@@ -68,11 +68,20 @@ class QueryRunnerTest: IntegrationTest() {
 
                 StudioState.pages.active?.let { if (it.isRunnable) it.asRunnable().mayOpenAndRun() }
 
+                val sessionIsData = StudioState.client.session.isData
+                val transactionIsRead = StudioState.client.session.transaction.transaction!!.type().isRead
+                val snapshotIsDisabled = !StudioState.client.session.transaction.snapshot.value
+
+                assert(sessionIsData)
+                assert(transactionIsRead)
+                assert(snapshotIsDisabled)
+
                 val priorTransaction = StudioState.client.session.transaction.transaction!!
```

Comment:
This is transaction that just took place, rather than a currently ongoing transaction. The session is not named in the same way as the session is ongoing.

Reply:
I see the awkwardness. It's actually quite coincidental that `StudioState...transaction` is even non-null at this point.

Let's make this test more robust by setting `snapshot` to true.

Reply:
and now our variable can just be `transaction`. Makes sense! Done.